src/app/pages/dashboard/services/widget-resources.service.ts

```diff
@@ -28,7 +37,34 @@ export class WidgetResourcesService {
     shareReplay({ bufferSize: 1, refCount: true }),
   );
 
+  readonly serverTime$ = this.store$.pipe(
+    waitForSystemInfo,
+    map((systemInfo) => new Date(systemInfo.datetime.$date)),
+    combineLatestWith(timer(0, 10000)),
+    map(([serverTime]) => {
+      serverTime.setSeconds(serverTime.getSeconds() + 10000 / 1000);
+      return serverTime;
+    }),
+  );
+
+  networkInterfaceUpdate(interfaceName: string): Observable<ReportingData[]> {
+    return this.serverTime$.pipe(
+      skipWhile(() => !interfaceName),
```

Comment:
Don't think this will work when `interfaceName` is null/empty string. string is passed by value so that variable will never update and this observable will keep skipping emits because of the `skipWhile`.
Also, I think computing time values on the spot are a better approach. We don't need this information all the time. Just when making the calls. Why keep timers on when they are not needed.

Reply:
Yeah, that's right. It looks like leftover since we are switching to `networkInterfaceUpdate` once the interface is loaded